### ISSUE TITLE
The PR #50334 breaks twister execution on various HW boards

### ISSUE THREAD

**Issue by evgeniy-paltsev** [zephyrproject-rtos/zephyr#51446]
**Describe the bug**

This is a regression, The PR #50334 breaks twister execution on various HW boards. The symptom is that we get test output rejected as it is treated as 'leftover' from previous test.

Here is an example - we run `hello_world` sample on `hsdk` board, (logs are truncated):

```
$ twister --jobs 2 -vvvv --ninja --platform hsdk --device-testing --device-serial /dev/serial-board-251642000154 --west-flash=--serial=251642000154 --test samples/hello_world/sample.basic.helloworld
    **[snip]**
DEBUG   - run test: hsdk/samples/hello_world/sample.basic.helloworld
DEBUG   - Using serial device /dev/serial-board-251642000154 @ 115200 baud
DEBUG   - Flash command: ['west', 'flash', '--skip-rebuild', '-d', '/home/ubuntu/workspace/zephyr/twister-out/hsdk/samples/hello_world/sample.basic.helloworld', '--', '--serial=251642000154']
DEBUG   - -- west flash: using runner openocd
L2 cache is present and not disabled
Invalidating L2 cache...
L2 cache has been flushed and invalidated.

DEBUG   - leftover log of previous test: b'*** Booting Zephyr OS build zephyr-v3.2.0-711-g4c53a4067a67  ***\r\n'
DEBUG   - leftover log of previous test: b'Hello World! hsdk\r\n'
DEBUG   - Timed out while monitoring serial output on hsdk
DEBUG   - run status: hsdk/samples/hello_world/sample.basic.helloworld error
INFO    - 1/1 hsdk                      samples/hello_world/sample.basic.helloworld        FAILED No Console Output(Timeout) (device 60.376s)
ERROR   - see: /home/ubuntu/workspace/zephyr/twister-out/hsdk/samples/hello_world/sample.basic.helloworld/device.log

INFO    - 0 of 1 test configurations passed (0.00%), 1 failed, 0 skipped with 0 warnings in 70.80 seconds
INFO    - In total 1 test cases were executed, 0 skipped on 1 out of total 508 platforms (0.20%)
INFO    - 1 test configurations executed on platforms, 0 test configurations were only built.
    **[snip]**
```

As we can see the test was successfully executed on board so we can see it's UART output: `Hello World! hsdk\r\n`
However test output was rejected as it was treated (incorrectly) as 'leftover' from previous test:

```
DEBUG   - leftover log of previous test: b'*** Booting Zephyr OS build zephyr-v3.2.0-711-g4c53a4067a67  ***\r\n'
DEBUG   - leftover log of previous test: b'Hello World! hsdk\r\n'
```

The bisecting blames 2cea5850e2cd7587858276da9d8c84ee90568824 commit which introduce this 'leftover' logic.
Reverting this commit fix the issue.

The full twister logs are attached: [hsdk_hello_world_fail_4c53a40.log](https://github.com/zephyrproject-rtos/zephyr/files/9821179/hsdk_hello_world_fail_4c53a40.log)

**To Reproduce**

You need to have HW board (hsdk in this example)

```
# checkout to current main
git checkout 4c53a4067a67
# use Zephyr SDK 0.15.0
export ZEPHYR_TOOLCHAIN_VARIANT=zephyr
export ZEPHYR_SDK_INSTALL_DIR=/home/ubuntu/zephyr-sdk/zephyr-sdk-0.15.0
# run twister. you may need to replace 251642000154 by your board serial number
twister --jobs 2 -vvvv --ninja --platform hsdk --device-testing --device-serial /dev/serial-board-251642000154 --west-flash=--serial=251642000154 --test samples/hello_world/sample.basic.helloworld
```

**Impact**

False positive test failure reports on HW platforms.

**Environment (please complete the following information):**
 - OS: Linux, Ubuntu 22.04.1 LTS, kernel version 5.15.0-50-generic
 - Toolchain: Zephyr SDK 0.15.0
 - Commit SHA: 4c53a4067a67
 - Bisected (root-cause) commit: 2cea5850e2cd7587858276da9d8c84ee90568824

**Comment by evgeniy-paltsev** [comment]
@smrtos - may I ask you to look at this as an author of https://github.com/zephyrproject-rtos/zephyr/commit/2cea5850e2cd7587858276da9d8c84ee90568824 ?

**Comment by evgeniy-paltsev** [comment]
Yep, https://github.com/zephyrproject-rtos/zephyr/pull/51467 fixes this, thanks!

**Comment by evgeniy-paltsev** [comment]
Fixed by #51467
